examples/cpp/BenchmarkHybridSearch.cpp

```diff
@@ -0,0 +1,164 @@
+    // target pointcloud = source.Transform(transformation_);
+    core::Tensor transformation_ =
+            core::Tensor::Init<float>({{0.862, 0.011, -0.507, 0.5},
+                                       {-0.139, 0.967, -0.215, 0.7},
+                                       {0.487, 0.255, 0.835, -1.4},
+                                       {0.0, 0.0, 0.0, 1.0}},
+                                      core::Device("CPU:0"));
+
+    t::geometry::PointCloud target_device(device);
+    t::geometry::PointCloud source_device(device);
+    PrepareInput(input_, transformation_, target_device, source_device, device,
+                 dtype);
+    core::Tensor init_trans = core::Tensor::Eye(4, dtype, device);
+    utility::LogInfo(" Processing Input on {} Success", device.ToString());
+
+    open3d::core::nns::NearestNeighborSearch target_nns(
+            target_device.GetPoints());
+    bool check = target_nns.HybridIndex(max_correspondence_dist);
+    if (!check) {
+        utility::LogError("Index Failed");
+    }
+
+    utility::LogInfo(" Source PointCloud size {}, Target PointCloud size {}",
+                     source_device.GetPoints().GetShape()[0],
+                     target_device.GetPoints().GetShape()[0]);
+
+    double avg_ = 0, max_ = 0, min_ = 0;
+    for (int i = 0; i < iterations; i++) {
+        utility::Timer hybrid_time;
+
+        // --- TIMER START
+        hybrid_time.Start();
+        auto result_nns = target_nns.HybridSearch(source_device.GetPoints(),
+                                                  max_correspondence_dist, 1);
+        hybrid_time.Stop();
+        // --- TIMER STOP
+
+        // To get number of correspondence
+        auto correspondence_set_ =
+                result_nns.first
+                        .IndexGet({result_nns.first.Ne(-1).Reshape({-1})})
+                        .Reshape({-1});
+        utility::LogInfo(" [Tensor] HYBRID SEARCH TOOK {}, Correspondences: {}",
+                         hybrid_time.GetDuration(),
+                         correspondence_set_.GetShape()[0]);
+
+        auto time = hybrid_time.GetDuration();
+        avg_ += time;
+        max_ = std::max(max_, time);
+        min_ = std::min(min_, time);
+    }
+    avg_ = avg_ / (double)iterations;
+
+    utility::LogInfo(" Average Time: {}, Max {}, Min {} ", avg_, max_, min_);
+
+    return 0;
+}
+
+inline void PrepareInput(t::geometry::PointCloud &input,
+                         core::Tensor &transformation_,
+                         t::geometry::PointCloud &target_device,
+                         t::geometry::PointCloud &source_device,
+                         core::Device device,
+                         core::Dtype dtype) {
+    // geometry::PointCloud legacy_s = source_.ToLegacyPointCloud();
+    geometry::PointCloud legacy_t = input.ToLegacyPointCloud();
+
+    // legacy_s.VoxelDownSample(voxel_downsample_factor);
+    legacy_t.VoxelDownSample(voxel_downsample_factor);
```

Comment:
`legacy_t = *legacy_t.VoxelDownSample(voxel_downsample_factor);` To downsample. 
Currently, it is not downsampling.

Reply:
Done